src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1679,8 +1687,11 @@ libtorrent::torrent_handle TorrentHandle::nativeHandle() const
 void TorrentHandle::updateTorrentInfo()
 {
     if (!hasMetadata()) return;
-
+#if LIBTORRENT_VERSION_NUM < 10100
     m_torrentInfo = TorrentInfo(m_nativeStatus.torrent_file);
+#else
+    m_torrentInfo = TorrentInfo(m_nativeStatus.torrent_file.lock());
```

Comment:
in the long run, I would recommend holding the torrent_info object by a shared_ptr instead of copying it. It can easily end up using a lot of memory otherwise